mintpy/closure_phase_bias.py

```diff
@@ -131,35 +672,156 @@ def main(iargs = None):
 
     # save mask
     meta = dict(stack_obj.metadata)
     meta['FILE_TYPE'] = 'mask'
     ds_name_dict = {'cpmask': [np.float32, (length, width), mask],}
-    writefile.layout_hdf5(os.path.join(inps.outdir,'cpmask.h5'), ds_name_dict, meta)
+    writefile.layout_hdf5(os.path.join(outdir,'cpmask.h5'), ds_name_dict, meta)
 
     # also save the average closure phase
     ds_name_dict2 = {'phase': [np.float32, (length, width), np.angle(closurephase)],
                     'amplitude':[np.float32,(length,width),np.abs(closurephase)/numcp],}
-    writefile.layout_hdf5(os.path.join(inps.outdir,'avgwcp.h5'), ds_name_dict2, meta)
+    writefile.layout_hdf5(os.path.join(outdir,'avgwcp.h5'), ds_name_dict2, meta)
+
+    return
+
+def compute_unwrap_closure_phase(ifgram_stack, conn, max_memory, outdir):
+    '''
+    Ouput wrapped, and unwrapped sequential closure phases, and cumulative closure phase time-series of connection-conn
+    '''
+    stack_obj = ifgramStack(ifgram_stack)
+    stack_obj.open()
+    length, width = stack_obj.length, stack_obj.width
+    meta = dict(stack_obj.metadata)
+    date12_list = stack_obj.get_date12_list(dropIfgram=True)
+    date12_list_all = stack_obj.get_date12_list(dropIfgram=False)
+    print('scene length, width', length, width)
+    ref_phase = stack_obj.get_reference_phase(unwDatasetName = 'unwrapPhase')
+    refX = stack_obj.refX
+    refY = stack_obj.refY
+    # retrieve the list of SLC dates from ifgramStack.h5
+    ifgram0 = date12_list[0]
+    date1, date2 = ifgram0.split('_')
+    slc_list = [date1, date2]
+    for ifgram in date12_list:
+        date1, date2 = ifgram.split('_')
+        if date1 not in slc_list:
+            slc_list.append(date1)
+        if date2 not in slc_list:
+            slc_list.append(date2)
+    slc_list.sort()
+    print('number of SLC found : ', len(slc_list))
+    print('first SLC: ', slc_list[0])
+    print('last  SLC: ', slc_list[-1])
+
+    # split igram_file into blocks to save memory
+    box_list, num_box = ifginv.split2boxes(ifgram_stack,max_memory)
+
+    closurephase =  np.zeros([len(slc_list)-conn, length,width],np.float32)
+    #process block-by-block
+    for i, box in enumerate(box_list):
+            box_width  = box[2] - box[0]
+            box_length = box[3] - box[1]
+            print(box)
+            if num_box > 1:
+                print('\n------- processing patch {} out of {} --------------'.format(i+1, num_box))
+                print('box width:  {}'.format(box_width))
+                print('box length: {}'.format(box_length))
+
+            closurephase[:,box[1]:box[3],box[0]:box[2]] = seq_closure_phase(slc_list, date12_list_all, ifgram_stack, ref_phase, conn, box)
+
+    # directory
+    cpdir = os.path.join(outdir, 'ClosurePhase')
+    if not os.path.isdir(cpdir):
+        os.mkdir(cpdir)
+
+    cpdir_conn = os.path.join(cpdir,'con'+str(conn)+'_cp')
```

Comment:
Both `con` and `conn` are currently used in the code for folder and file names, could you ensure all of them are using `conn` only for consistency?

Reply:
Thanks! Modified as suggested.